### PR TITLE
chore(CSVParser): migrate parser code into UI codebase

### LINKED ISSUE #5662: csv upload experience: translate non-annotated csv parsing into JavaScript
### Background
The telegraf codebase has a [CSV parser](https://github.com/influxdata/telegraf/blob/master/plugins/parsers/csv/parser.go) written in go that handles annotated and non-annotated CSVs. Take that code's structure, and translate it from go into JavaScript. This will require following the code, looking up library functions in go like `strings.SplitN()`, and either using the analagous JavaScript function, or writing similar code.

### Acceptance Criteria
- Translate the CSV go parser into JavaScript
   - The parser should handle annotated and un-annotated CSVs
- Test the parser with unit tests to confirm that it parses un-annotated CSVs properly

### PR DESCRIPTION
**Pull request by blegesse-git**
Closes #5662 

As part of DND non-annotated CSV epic, pr introduces CSV parsing code that will be integrated into the UI to allow users the ability to upload non-annotated CSVs into the UI.

`parser.ts` is a direct translation of telegraf's [CSV parser](https://github.com/influxdata/telegraf/blob/master/plugins/parsers/csv/parser.go). 
This javascript parser leverages an external library, Papa Parse to read and parse csv files.

Upcoming work utilizing this parser will be hidden behind a feature flag `uploadCSV`. 

### Checklist

Authors and Reviewer(s), please verify the following:

- [x] A PR description, regardless of the triviality of this change, that communicates the value of this PR
- [x] [Well-formatted conventional commit messages](https://www.conventionalcommits.org/en/v1.0.0/) that provide context into the change
- [x] Documentation updated or issue created (provide link to issue/PR)
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [ ] Feature flagged, if applicable

### CODE REVIEW

File: src/utils/csvParser/parser.ts
Comment (hoorayimhelping) [inline src/utils/csvParser/parser.ts:7]: before you merge this in, let's pair up and see if we can do this without adding a new dependency

Comment (blegesse-git) [reply src/utils/csvParser/parser.ts:7]: yup! sorry, forgot we said we'd do that first

File: src/utils/csvParser/parser.test.ts
Comment (hoorayimhelping) [inline src/utils/csvParser/parser.test.ts:1]: how many eslint errors are there related to loss of precision?

File: src/utils/csvParser/parser.test.ts
Comment (hoorayimhelping) [inline src/utils/csvParser/parser.test.ts:2]: is this import necessary or can it be done away with? it doesn't look like any of our other tests import `expect` or `jest`

File: src/utils/csvParser/parser.test.ts
Comment (hoorayimhelping) [inline src/utils/csvParser/parser.test.ts:6]: rather than re-creating this here and in `parser.ts`, just export it from `parser.ts` and import it here

File: src/utils/csvParser/parser.ts
Comment (hoorayimhelping) [inline src/utils/csvParser/parser.ts:8]: is it possible to reuse [`Point`](https://github.com/influxdata/ui/blob/master/src/cloud/apis/reporting.ts#L6) from the reporting section of the app?

File: src/utils/csvParser/parser.ts
Comment (hoorayimhelping) [inline src/utils/csvParser/parser.ts:160]: what's this comment mean? I think it's okay to remove

File: src/utils/csvParser/parser.ts
Comment (hoorayimhelping) [inline src/utils/csvParser/parser.ts:214]: can/should this be something like

```ts
const metrics: Metric[] = await ...
```
?

**Review by hoorayimhelping — COMMENTED** [review]
looks great! just a couple of minor nagging thoughts - let me know what you think

File: src/utils/csvParser/parser.test.ts
Comment (blegesse-git) [reply src/utils/csvParser/parser.test.ts:1]: Just one error. 
precision_loss_screenshot
Instead of disabling eslint for the entire file, I can just disable it for this specific line.

File: src/utils/csvParser/parser.test.ts
Comment (blegesse-git) [reply src/utils/csvParser/parser.test.ts:2]: you're right. Other unit tests in the code base don't import jest 🤔. But removing the import throws a lot of errors like this: 
jest_error
I can't think of why this test file requires an import while others don't. I'll look more into this

File: src/utils/csvParser/parser.test.ts
Comment (blegesse-git) [reply src/utils/csvParser/parser.test.ts:6]: good call

File: src/utils/csvParser/parser.ts
Comment (blegesse-git) [reply src/utils/csvParser/parser.ts:214]: we can. But because `this.parseCSV()` is returning metrics with explicit `Metric` type, typescript is able to make that inference. But i don't see the harm in adding it here.

File: src/utils/csvParser/parser.ts
Comment (blegesse-git) [reply src/utils/csvParser/parser.ts:8]: Yes! I ended up importing it and using that instead. Had no clue it already existed in the codebase.

File: src/utils/csvParser/parser.test.ts
Comment (hoorayimhelping) [reply src/utils/csvParser/parser.test.ts]: I wonder if this is the first test to be created after we upgraded jest

**Review by hoorayimhelping — APPROVED** [review]

**Pull request merged**
